presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -320,19 +336,25 @@ private void validateLocalityGroups(ConnectorTableMetadata meta)
 
         // For each locality group
         for (Map.Entry<String, Set<String>> g : groups.get().entrySet()) {
-            // For each column in the group
-            for (String col : g.getValue()) {
-                // If the column was not found, throw exception
-                List<ColumnMetadata> matched = meta.getColumns().stream()
-                        .filter(x -> x.getName().equalsIgnoreCase(col)).collect(Collectors.toList());
+            if (g.getValue().contains(rowIdColumn)) {
+                throw new PrestoException(VALIDATION, "Row ID column cannot be in a locality group");
+            }
 
-                if (matched.size() != 1) {
-                    throw new PrestoException(VALIDATION, "Unknown column in locality group: " + col);
+            // Validate the specified column names exist in the table definition,
+            // incrementing a counter for each matching column
+            int matchingColumns = 0;
+            for (ColumnMetadata column : meta.getColumns()) {
+                if (g.getValue().contains(column.getName().toLowerCase(Locale.ENGLISH))) {
+                    ++matchingColumns;
                 }
+            }
 
-                if (matched.get(0).getName().equalsIgnoreCase(rowIdColumn)) {
-                    throw new PrestoException(VALIDATION, "Row ID column cannot be in a locality group");
-                }
+            // If the number of matched columns does not equal the defined size,
+            // then a column was specified that does not exist
+            // (or there is a duplicate column in the table DDL, which is also an issue but
+            // has been checked before in validateColumns)
+            if (matchingColumns != g.getValue().size()) {
```

Comment:
You can move it into the loop (https://github.com/prestodb/presto/pull/5030/files/65836f8f880c493f2cbb605fd4142d0a67cf54fa..8869b1300b345ff6e4c7cf69929283ddc5968440#diff-68cb92636ee64a829fa0a3578bf988f7R348), to stop looping as soon as condition is met.